pkg/stores/sqlproxy/proxy_store.go

```diff
@@ -676,53 +533,43 @@ func (s *Store) ListByPartitions(apiOp *types.APIRequest, schema *types.APISchem
 		return nil, 0, "", err
 	}
 
-	return list.Items, total, continueToken, nil
+	return list, total, continueToken, nil
 }
 
 // WatchByPartitions returns a channel of events for a list or resource belonging to any of the specified partitions
-func (s *Store) WatchByPartitions(apiOp *types.APIRequest, schema *types.APISchema, wr types.WatchRequest, partitions []partition.Partition) (chan watch.Event, error) {
-	ctx, cancel := context.WithCancel(apiOp.Context())
-	apiOp = apiOp.Clone().WithContext(ctx)
-
-	eg := errgroup.Group{}
-
-	result := make(chan watch.Event)
+func (s *Store) WatchByPartitions(apiOp *types.APIRequest, schema *types.APISchema, wr types.WatchRequest, partitions []partition.Partition) (chan struct{}, error) {
+	ctx := apiOp.Context()
 
-	for _, partition := range partitions {
-		p := partition
-		eg.Go(func() error {
-			defer cancel()
-			c, err := s.watchByPartition(p, apiOp, schema, wr)
-
-			if err != nil {
-				return err
-			}
-			for i := range c {
-				result <- i
-			}
-			return nil
-		})
+	revision := 0
+	if wr.Revision != "" {
+		parsedRevision, err := strconv.ParseInt(wr.Revision, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("invalid revision %q: %w", wr.Revision, err)
+		}
+		revision = int(parsedRevision)
 	}
 
-	go func() {
-		defer close(result)
-		<-ctx.Done()
-		eg.Wait()
-		cancel()
-	}()
+	// XXX: Why was this needed at all??
+	apiOp = apiOp.Clone().WithContext(ctx)
```

Comment:
This was copied over from the non-SQLite implementation of Watch, which has this line:

https://github.com/rancher/steve/blob/99e479ba0f0883bf7337b3eba8a67200dcc7509c/pkg/stores/partition/store.go#L295

That seems to come from a distant past:

https://github.com/rancher/steve/commit/a9c39ef89b26ad9ef52d9fdfef5249da894ad7ad#diff-74cdf37ae3cf46d709e040c4c6e22a30e4b62586165b5f60c71f2e3f02514d98R132

The only reason I can see is we do not want to change the passed in `apiOp`, as it is passed by pointer. As Stores can be nested, I can see a calling Store not to expect the called Store to modify the passed `apiOp` to add a context to the nested `Request` the `apiOp` wraps.

So my understanding is that cloning here has the effect to have the context be applied to callees exclusively, but not callers. Whether this really prevents any bugs I am not sure.